### PR TITLE
SUIT support

### PR DESCRIPTION
**Pull request by d3zd3z**
This is mostly completed work to add SUIT support to the simulator. This does not add SUIT support to MCUboot itself, so what this mostly does is create a new feature "suit-manifest" that can be enabled to cause test failures.

### CODE REVIEW

**Comment by d3zd3z** [comment]
Note that github doesn't show the commits in a meaningful order. Probably best to fetch this with `git fetch origin pull/391/head` and look at the changes locally.

**Comment by d3zd3z** [comment]
One big question I have for reviewers is: should I squash the cbor and/or suit commits. There is a bit of development work in here, and these could make sense just having a commit that adds them.

File: sim/cboria/src/bin/cbor-dump.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    Ok(())
+}
+
+/// Read a CBOR code value, and the associated integer.
+fn get_code<R: Read>(rd: &mut R) -> Result<(u8, usize)> {
+    let b = rd.read_u8()?;
+    let code = b >> 5;
+    let count = match b & 0x1f {
+        count @ 0 ..= 23 => count as usize,
```
Comment (utzig) [inline sim/cboria/src/bin/cbor-dump.rs]: The binding should be `count @ 0 ... 23`, no? What does the `..=` syntax mean?

Comment (d3zd3z) [reply sim/cboria/src/bin/cbor-dump.rs]: `..=` is an inclusive range pattern (as opposed to `0 .. 24` which would be the same range). This is new to Rust 2018.

**Comment by utzig** [comment]
@d3zd3z After reading through some of the cboria commits, it looks like it would be easier to just squash those commits (I just want to review the final version, not the progress of making it!).

**Comment by d3zd3z** [comment]
I've squashed the cbor/suit commits. Now most of the commits are modifications to the simulator to call this code.

File: boot/bootutil/src/cbor.c [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    } else if (ext == 24) {
+        want = 1;
+    } else if (ext == 25) {
+        want = 2;
+    } else if (ext == 26) {
+        want = 4;
+    } else {
+        /* If we want uint64 support, add it above here. */
+        return -1;
+    }
```
Comment (utzig) [inline boot/bootutil/src/cbor.c]: *maybe* would gain a few bytes by doing:
```
    ...
    return 0;
} else if (ext > 26) {
    return -1;
}

want = 1 << (ext - 24);
```

Comment (d3zd3z) [reply boot/bootutil/src/cbor.c]: 10-12 bytes. 10 bytes of code difference, possibly causing 12 bytes of padding.

**Comment by d3zd3z** [comment]
I have moved the simulator changes to the front, and pushed to #434. If desired, I can even remove them from this patch series (just a little tweaking with the capabilities flags), perhaps pushing that change through, and these can merge independently.

**Comment by d3zd3z** [comment]
Simulator changes moved into #435. Merge these two changes to be able to test the SUIT changes within the simulator.

**Comment by utzig** [comment]
Depends on #435

**Comment by d3zd3z** [comment]
This has gotten quite out of date, and SUIT has moved to a new version. The pull request will remain, even when closed, but I'm closing this to reduce clutter of the pull request list.

**Pull request closed without merging**